### PR TITLE
Added DAE solvers

### PR DESCRIPTION
**Pull request by ParamThakkar123**
## Checklist

- [X] Appropriate tests were added
- [X] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [X] Any new documentation only uses public API
  
## Additional context

solves a part of #2177

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Looks like this has merge issues. And the DAE things here are just BDF. This should just be combined with the other bdf

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas Then should I put all the DAE algorithms with the BDF ones ??

**Comment by ChrisRackauckas** [comment]
Yes

File: lib/OrdinaryDiffEqBDF/src/algorithms.jl [hunk 1]
```diff
@@ -388,3 +388,79 @@ unew = uold + dt * (f1(unew) + f2(y1))
...
+  nlsolve::F2
+  precs::P
+  extrapolant::Symbol
+end
+
+DBDF(;chunk_size=Val{0}(),autodiff=Val{true}(), standardtag = Val{true}(), concrete_jac = nothing,diff_type=Val{:forward},
+     linsolve=nothing,precs = DEFAULT_PRECS,nlsolve=NLNewton(),extrapolant=:linear) =
+     DBDF{_unwrap_val(chunk_size),_unwrap_val(autodiff),typeof(linsolve),typeof(nlsolve),typeof(precs),diff_type,_unwrap_val(standardtag),_unwrap_val(concrete_jac)}(
+     linsolve,nlsolve,precs,extrapolant)
+=#
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqBDF/src/algorithms.jl:441]:
```suggestion
```

**Comment by ChrisRackauckas** [comment]
Rebase this onto master

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas There's a `compute_step!()` method failure occurring here. How can we deal with this ??

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas I started a new PR for QPRK solvers #2301

**Pull request merged**